x-pack/plugins/cases/server/client/cases/update_custom_field.ts

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import Boom from '@hapi/boom';
+
+import type { CasesClient, CasesClientArgs } from '..';
+
+import type {
+  CaseRequestCustomField,
+  CaseRequestCustomFields,
+  CustomFieldPatchRequest,
+} from '../../../common/types/api';
+import { CustomFieldsRt } from '../../../common/types/api';
+import { Operations } from '../../authorization';
+import { createCaseError } from '../../common/error';
+import { flattenCaseSavedObject } from '../../common/utils';
+import { decodeOrThrow } from '../../../common/api/runtime_types';
+import type { Case } from '../../../common/types/domain';
+import { CaseRt } from '../../../common/types/domain';
+import { decodeWithExcessOrThrow } from '../../../common/api';
+import {
+  validateCustomFieldKeysAgainstConfiguration,
+  validateCustomFieldTypesInRequest,
+} from './validators';
+import type { UserActionEvent } from '../../services/user_actions/types';
+
+export interface UpdateCustomFieldArgs {
+  /**
+   * The ID of a case
+   */
+  caseId: string;
+  /**
+   * The ID of a custom field to be updated
+   */
+  customFieldId: string;
+  /**
+   * value of custom field to update, case version
+   */
+  request: CustomFieldPatchRequest;
+}
+
+/**
+ * Updates the specified cases with new values
+ *
+ * @ignore
+ */
+export const updateCustomField = async (
+  { caseId, customFieldId, request }: UpdateCustomFieldArgs,
+  clientArgs: CasesClientArgs,
+  casesClient: CasesClient
+): Promise<Case> => {
+  const {
+    services: { caseService, userActionService },
+    user,
+    logger,
+    authorization,
+  } = clientArgs;
+
+  try {
+    const { value, caseVersion } = request;
+
+    const caseToUpdate = await caseService.getCase({
+      id: caseId,
+    });
+
+    const configurations = await casesClient.configure.get({
+      owner: caseToUpdate.attributes.owner,
+    });
+
+    await authorization.ensureAuthorized({
+      entities: [{ owner: caseToUpdate.attributes.owner, id: caseToUpdate.id }],
+      operation: Operations.updateCase,
+    });
+
+    const foundCustomField = configurations[0]?.customFields.find(
+      (item) => item.key === customFieldId
+    );
+
+    if (!foundCustomField) {
+      throw Boom.badRequest('cannot find custom field');
+    }
+
+    const decodedCustomFields = decodeWithExcessOrThrow(CustomFieldsRt)([
+      {
+        value,
+        type: foundCustomField.type,
+        key: customFieldId,
+      },
+      ...caseToUpdate.attributes.customFields.filter((field) => field.key !== customFieldId),
+    ]);
+
+    const updatedAt = new Date().toISOString();
+
+    if (value == null && foundCustomField.required) {
```

Comment:
What if the value is `'       '` or `undefined`?

Reply:
hmm good point. should also check for empty value with trim